pkg/client/config/config.go

```diff
@@ -29,15 +29,30 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/internal/log"
 )
 
+// KubeconfigFlagName is the name of the kubeconfig flag
+const KubeconfigFlagName = "kubeconfig"
+
 var (
 	kubeconfig string
 	log        = logf.RuntimeLog.WithName("client").WithName("config")
 )
 
+// init registers the "kubeconfig" flag to the default command line FlagSet.
+// TODO: This should be removed, as it potentially leads to redefined flag errors for users, if they already
+// have registered the "kubeconfig" flag to the command line FlagSet in other parts of their code.
 func init() {
-	// TODO: Fix this to allow double vendoring this library but still register flags on behalf of users
-	flag.StringVar(&kubeconfig, "kubeconfig", "",
-		"Paths to a kubeconfig. Only required if out-of-cluster.")
+	RegisterFlags(flag.CommandLine)
+}
+
+// RegisterFlags registers flag variables to the given FlagSet if not already registered.
+// It uses the default command line FlagSet, if none is provided. Currently, it only registers the kubeconfig flag.
+func RegisterFlags(fs *flag.FlagSet) {
+	if fs == nil {
+		fs = flag.CommandLine
+	}
+	if fs.Lookup(KubeconfigFlagName) == nil {
+		fs.StringVar(&kubeconfig, KubeconfigFlagName, "", "Paths to a kubeconfig. Only required if out-of-cluster.")
```

Comment:
But if the flag is registered and ppl attempt to use it, it will just silently not work or not?

Reply:
You are right. In the case you describe the value is not stored in the `kubeconfig` string var. We could solve this by getting the value if it was already set:
```go
	f := fs.Lookup(KubeconfigFlagName)
	if f != nil {
		kubeconfig = f.Value.String()
	} else {
		fs.StringVar(&kubeconfig, KubeconfigFlagName, "", "Paths to a kubeconfig. Only required if out-of-cluster.")
	}
```

Or should we simply drop the lookup and expect that there is not flag yet provided if someone is about to call `RegisterFlags`?

Reply:
Looking it up would be great if it already exists.

> Or should we simply drop the lookup and expect that there is not flag yet provided if someone is about to call RegisterFlags?

No super strong opinion here, but in case of someone explicitly calling `RegisterFlags` I would not expect it to be defined. In the case of the `init` though it might be

Reply:
Sorry for my late reply. I pushed a new commit containing the lookup.